libdnf/sack/query.cpp

```diff
@@ -769,6 +831,116 @@ Query::addFilter(HyNevra nevra, bool icase)
     return 0;
 }
 
+void
+Query::Impl::filterNevraStrict(int cmpType, const char **matches)
```

Comment:
This function is too long, too messy to check. I have no idea what it does.

Reply:
Shortened